chapters/implementation.tex

```diff
@@ -1,3 +1,54 @@
 \section{Implementation} \label{sec:implementation}
-Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren.
 
+For the implementation of the app, Kotlin was used.
+
+Kotlin is ``[...] an open-source statically typed programming language
+that targets the JVM, Android, JavaScript and Native.'' \cite{kotlin2020}
+Google officially supports Kotlin for Android Apps since 2017
+\cite{googleio2017} and recommends using Kotlin since 2019
+\cite{androidkotlin2019}.
+
+Thanks to the similarity with languages like Java and TypeScript, adapting
+Kotlin as programming language was relatively easy for the project team.
+
+Various language features of Kotlin helped in development, too. Instead of
+implementing the singleton pattern, Kotlin allows to create an single object
+per runtime by using the \texttt{object}-Keyword.
+
+Some design choices of the app were slowed down by Kotlins language features.
+As all Kotlin classes inherit from \texttt{Any}, using features of Java's 
+\texttt{Object}-class are not natively available. This made using the methods
+\texttt{.wait()} and \texttt{.notify()}, which are used for synchronisation,
+more difficult.
+
+Using ``Android Studio'', the official IDE for Android development, allowed
```

Comment:
I'd write `Android Studio, known as the official IDE for Android development, allowed ...`